firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigContainerTest.java

```diff
@@ -138,23 +144,150 @@ public void getChangedParams_sameP13nMetadata_returnsEmptySet() throws Exception
+    JSONArray fetchedExperiments = new JSONArray();
+
+    ConfigContainer config =
+        ConfigContainer.newBuilder()
+            .replaceConfigsWith(ImmutableMap.of("abt_test_key_1", "value_1"))
+            .withAbtExperiments(activeExperiments)
+            .build();
+
+    ConfigContainer other =
+        ConfigContainer.newBuilder()
+            .replaceConfigsWith(ImmutableMap.of("abt_test_key_1", "value_1"))
+            .withAbtExperiments(fetchedExperiments)
+            .build();
+
+    Set<String> changedParams = config.getChangedParams(other);
+
+    assertThat(changedParams).containsExactly("abt_test_key_1");
+  }
+
+  @Test
+  public void getChangedParams_changedExperimentsKeys_returnsUpdatedKey() throws Exception {
+    JSONArray activeExperiments = generateAbtExperiments(1);
+    JSONArray fetchedExperiments = generateAbtExperiments(1);
+
+    fetchedExperiments.getJSONObject(0).getJSONArray(AFFECTED_PARAMETER_KEYS).put("abt_test_key_2");
+
+    ConfigContainer config =
+        ConfigContainer.newBuilder()
+            .replaceConfigsWith(
+                ImmutableMap.of("abt_test_key_1", "value_1", "abt_test_key_2", "value_2"))
+            .withAbtExperiments(activeExperiments)
+            .build();
+
+    ConfigContainer other =
+        ConfigContainer.newBuilder()
+            .replaceConfigsWith(
+                ImmutableMap.of("abt_test_key_1", "value_1", "abt_test_key_2", "value_2"))
+            .withAbtExperiments(fetchedExperiments)
+            .build();
+
+    Set<String> changedParams = config.getChangedParams(other);
+
+    assertThat(changedParams).containsExactly("abt_test_key_2");
+  }
+
+  @Test
+  public void getChangedParams_changedExperimentsStartTime_returnsUpdatedKey() throws Exception {
+    JSONArray activeExperiments = generateAbtExperiments(1);
+    JSONArray fetchedExperiments = generateAbtExperiments(1);
+
+    fetchedExperiments.getJSONObject(0).put(EXPERIMENT_START_TIME, new Date(2089));
+
+    ConfigContainer config =
+        ConfigContainer.newBuilder()
+            .replaceConfigsWith(ImmutableMap.of("abt_test_key_1", "value_1"))
+            .withAbtExperiments(activeExperiments)
+            .build();
+
+    ConfigContainer other =
+        ConfigContainer.newBuilder()
+            .replaceConfigsWith(ImmutableMap.of("abt_test_key_1", "value_1"))
+            .withAbtExperiments(fetchedExperiments)
+            .build();
+
+    Set<String> changedParams = config.getChangedParams(other);
+
+    assertThat(changedParams).containsExactly("abt_test_key_1");
+  }
+
+  @Test
+  public void getChangedParams_changedExperimentsNoTriggerEvent_returnsUpdatedKey()
+      throws Exception {
+    JSONArray activeExperiments = generateAbtExperiments(1);
+    JSONArray fetchedExperiments = generateAbtExperiments(1);
+
+    fetchedExperiments.getJSONObject(0).remove(TRIGGER_EVENT);
+
+    fetchedExperiments.getJSONObject(0).put(EXPERIMENT_START_TIME, new Date(2089));
```

Comment:
is EXPERIMENT_START_TIME significant here? maybe copy-pasted from the previous test?

Reply:
It's just to simulate ExperimentDescription changes or any type but I removed it.